### PR TITLE
Add test for config/validation case sensitivity and allow nested elements of different case

### PR DESCRIPTION
**Pull request by alexsm82**
Given nested elements are not case-sensitive, add tests for nested elements with a case that does not match what is specified in schema.

### CODE REVIEW

File: dev/com.ibm.ws.rest.handler.config_fat/fat/src/com/ibm/ws/rest/handler/config/fat/ConfigRESTHandlerTest.java [hunk 1]
```diff
@@ -452,62 +478,6 @@ public void testConfigDataSource() throws Exception {
         assertEquals(err, true, j.getBoolean("transactional"));
         assertNotNull(err, j = j.getJsonObject("properties.derby.embedded"));
         assertEquals(err, "memory:doesNotExist", j.getString("databaseName"));
-
-        j = json.getJsonObject(5);
-        assertEquals(err, "dataSource", j.getString("configElementName"));
-        assertEquals(err, "transaction/dataSource[default-0]", j.getString("uid"));
```
Comment (njr-11) [inline dev/com.ibm.ws.rest.handler.config_fat/fat/src/com/ibm/ws/rest/handler/config/fat/ConfigRESTHandlerTest.java]: This is our only test of config output for a dataSource nested under a singleton pid.  Do not remove it.

File: dev/com.ibm.ws.rest.handler.validator_fat/fat/src/com/ibm/ws/rest/handler/validator/fat/ValidateDataSourceTest.java [hunk 2]
```diff
@@ -304,22 +304,6 @@ public void testMultiple() throws Exception {
         assertTrue(err, stack.getString(0).startsWith("org.apache.derby."));
         assertTrue(err, stack.getString(1).startsWith("org.apache.derby."));
         assertTrue(err, stack.getString(2).startsWith("org.apache.derby."));
-
-        // [5]: config.displayId=transaction/dataSource[default-0]
```
Comment (njr-11) [inline dev/com.ibm.ws.rest.handler.validator_fat/fat/src/com/ibm/ws/rest/handler/validator/fat/ValidateDataSourceTest.java]: Same comment as previously. Do not remove the only tests of dataSource nested under singleton pid.

File: dev/com.ibm.ws.rest.handler.validator_fat/fat/src/com/ibm/ws/rest/handler/validator/fat/ValidateDataSourceTest.java [hunk 3]
```diff
@@ -329,22 +313,6 @@ public void testMultipleWithNoResults() throws Exception {
         assertEquals("unexpected response: " + json, 0, json.size());
     }
 
-    @Test
-    public void testNestedUnderTransaction() throws Exception {
-        JsonObject json = new HttpsRequest(server, "/ibm/api/validation/dataSource/transaction%2FdataSource[default-0]?auth=container").run(JsonObject.class);
```
Comment (njr-11) [inline dev/com.ibm.ws.rest.handler.validator_fat/fat/src/com/ibm/ws/rest/handler/validator/fat/ValidateDataSourceTest.java]: Same as previous comments - do not remove the only tests of dataSource nested under singleton pid.

**Comment by alexsm82** [comment]
#build

**Comment by alexsm82** [comment]
#build

**Comment by alexsm82** [comment]
The build failures are existing, unrelated defects.

**Pull request merged**